### ISSUE TITLE
Gallery UI is extremely slow

### ISSUE THREAD

**Issue by kirgene** [flutter/gallery#896]
I've installed [Flutter Gallery](https://play.google.com/store/apps/details?id=io.flutter.demo.gallery) app on my Android phone and noticed that animations, scrolling and transitions are lagging.

Interesting observation: often once particular animation was performed it not longer lags. Same applies to scrolling: once I scroll to the end of the list scrolling starts working fine.

Seems that there is some kind of cache that is populated and only after that acceptable performance is achieved.

Here is a demo:
https://drive.google.com/open?id=1ZXKLWQEYLpOcjd5h7ndrMqHhC1GW37Rg
Notice how slow is tap animation when I press the button first time.
Also the last screen (Contact Profile) is unusable: scrolling and overall performance is extremely slow.

I've also tested this app on Android Tablet and got the same behavior.

**Phone: Samsung Galaxy S4 (GT-I9500)**
**Tablet: ASUS MeMO Pad FHD 10 (ME302C)**

**Comment by guidezpl** [comment]
Given this issue is 4 years old, and the app has seen many performance improvements (as well as Flutter itself, with Impeller for example), I'll close this issue. 

If you encounter performance issues (e.g. janky frames), please file a new issue with relevant information.
